### PR TITLE
feat: save and continue

### PR DESCRIPTION
**Pull request by jaredcwhite**
This PR addresses #3654 

- [x] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

## Description

This PR changes the "Save & Exit" button so it just reads "Save", and clicking it will save the listing and remain in the edit state. In addition, the "Cancel" button has been renamed "Exit" and it displays a prompt to the user to either save or discard changes.

## How Can This Be Tested/Reviewed?

Try saving or exiting listings in various stages (new, unpublished, published, etc.) and verify the behavior matches the AC/Figma.

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs
- [x] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [x] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

### CODE REVIEW

File: sites/partners/cypress/e2e/default/03-listing.spec.ts [hunk 1]
```diff
@@ -49,6 +49,26 @@ describe("Listing Management Tests", () => {
...
+    cy.getByID("listingsExitButton").click()
+    cy.getByID("listing-save-before-exit-dialog-content").contains(
+      "Do you want to save your changes before you exit?"
+    )
+    cy.getByID("saveBeforeExitDiscard").click()
+    cy.contains("Test - error messaging")
+    cy.getByID("listingEditButton").contains("Edit").click()
+    cy.getByID("name").clear()
+    cy.getByID("name").type("Test - error messaging DISCARD")
+    cy.getByID("listingsExitButton").click()
```
Comment (emilyjablonski) [inline sites/partners/cypress/e2e/default/03-listing.spec.ts:65]: Could we use getByText for selectors like this where possible?

File: sites/partners/src/components/listings/ListingFormActions.tsx [hunk 2]
```diff
@@ -74,14 +78,19 @@ const ListingFormActions = ({
   }, [listing])
 
   const actions = useMemo(() => {
+    // TODO: should be Exit, show Save confirmation dialog
```
Comment (emilyjablonski) [inline sites/partners/src/components/listings/ListingFormActions.tsx:81]: Is this TODO valid?

File: sites/partners/src/components/listings/ListingFormActions.tsx [hunk 3]
```diff
@@ -314,8 +324,9 @@ const ListingFormActions = ({
           type="button"
           className="w-full"
           onClick={() => {
-            // TODO throw a modal
-            submitFormWithStatus(true, ListingsStatusEnum.active)
+            // NOTE: confused by the following todo. Doesn't this show a modal? -JW
+            // TODO: throw a modal -EJ
```
Comment (emilyjablonski) [inline sites/partners/src/components/listings/ListingFormActions.tsx:328]: Another q on the TODO here

File: sites/partners/src/components/listings/PaperListingForm/dialogs/CloseListingDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import React from "react"
```
Comment (emilyjablonski) [inline sites/partners/src/components/listings/PaperListingForm/dialogs/CloseListingDialog.tsx:1]: I like these living in this separate folder!

**Comment by emilyjablonski** [comment]
Screenshot 2024-08-22 at 11 34 47 AM
When I get a backend error to throw I'm still getting the success toast

**Comment by emilyjablonski** [comment]
Screenshot 2024-08-22 at 11 40 15 AM
I feel like the behavior here is preferred, but if a listing is already live and I click save, I get this modal and then get redirected to the edit page instead of this new save flow.

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Would love to fix the listing name bug in particular but the rest are more questions / thoughts!

**Comment by jaredcwhite** [comment]
Thanks for the review @emilyjablonski! Working through the various issues/questions now.

File: sites/partners/src/components/listings/ListingFormActions.tsx (hunk 2 above)
Comment (jaredcwhite) [reply sites/partners/src/components/listings/ListingFormActions.tsx]: will delete!

File: sites/partners/cypress/e2e/default/03-listing.spec.ts (hunk 1 above)
Comment (jaredcwhite) [reply sites/partners/cypress/e2e/default/03-listing.spec.ts:65]: Hmm, well for a simple string like "Exit" I'm not sure if we want to assume there's only one "Exit" on the page…?

**Comment by jaredcwhite** [comment]
Just pushed up fixes for the listing name in header not changing and the error alert not disappearing.

**Comment by jaredcwhite** [comment]
> Screenshot 2024-08-22 at 11 40 15 AM
> I feel like the behavior here is preferred, but if a listing is already live and I click save, I get this modal and then get redirected to the edit page instead of this new save flow - but I also think that makes sense, I just don't see it covered in the AC.

I think the AC was unspecified with regard to confirmation modals. I think this makes sense too, but we could ask for design QA to verify.

**Review by emilyjablonski — APPROVED** [review]
🛹

**Pull request merged**